src/main/index.ts

```diff
@@ -299,49 +376,40 @@ app.on('quit', () => {
   }
 });
 
-// app.on("activate", () => {
-//   // On OS X it's common to re-create a window in the app when the
-//   // dock icon is clicked and there are no other windows open.
-//   if (BrowserWindow.getAllWindows().length === 0) {
-//     createMainWindow();
-//   }
-// });
-
-// const contextMenu = Menu.buildFromTemplate([
-//   {
-//     label: 'Open',
-//     type: 'normal',
-//     click() {
-//       if (SPLASH_SCREEN_WINDOW) {
-//         SPLASH_SCREEN_WINDOW.show();
-//       } else if (MAIN_WINDOW) {
-//         MAIN_WINDOW.show();
-//       }
-//     },
-//   },
-//   {
-//     label: 'Restart',
-//     type: 'normal',
-//     click() {
-//       const options: Electron.RelaunchOptions = {
-//         args: process.argv,
-//       };
-//       // https://github.com/electron-userland/electron-builder/issues/1727#issuecomment-769896927
-//       if (process.env.APPIMAGE) {
-//         console.log('process.execPath: ', process.execPath);
-//         options.args?.unshift('--appimage-extract-and-run');
-//         options.execPath = process.env.APPIMAGE;
-//       }
-//       app.relaunch(options);
-//       app.quit();
-//     },
-//   },
-//   {
-//     label: 'Quit',
-//     type: 'normal',
-//     click() {
-//       app.quit();
-//     },
-//   },
-// ]);
-// });
+/**
+ * This handler will make sure that the main window only gets hidden instead of
+ * closed (to maintain the javascript state) if the systray icon option is
+ * used.
+ *
+ * @param e Window close event
+ */
+const mainWindowCloseHandler = (e: Event) => {
+  if (!IS_APP_QUITTING && MAIN_WINDOW) {
+    e.preventDefault();
+    MAIN_WINDOW.hide();
+
+    const notificationIcon = nativeImage.createFromPath(NOTIFICATIONS_ICON_PATH);
+    new Notification({
+      title: `${KANGAROO_CONFIG.productName} keeps running in the background`,
+      body: `To close ${KANGAROO_CONFIG.productName} and stop synching with peers, quit from the icon in the system tray.`,
+      icon: notificationIcon,
+    })
+      .on('click', async () => {
+        if (MAIN_WINDOW) {
+          MAIN_WINDOW.show();
+          const response = await dialog.showMessageBox(MAIN_WINDOW, {
+            type: 'info',
+            message: `${KANGAROO_CONFIG.productName} keeps running in the background if you close the Window.\n\nThis is to keep synchronizing data with peers.\n\nDo you want to quit ${KANGAROO_CONFIG.productName} fully?`,
+            buttons: ['Keep Running', 'Quit'],
+            defaultId: 0,
+            cancelId: 1,
+          });
+          if (response.response === 1) {
+            app.quit();
```

Comment:
Ah, looks like quit is really just closing the main window?

Reply:
No quit is actually fully quitting the app, unless am listener to the `before-quit` or `will-quit` events is registered that would prevent it.

Reply:
So if the app is quit, is Holochain still running?

Reply:
No, holochain is running as a child process of the main (electron) nodejs process and also gets explicitly stopped [here](https://github.com/holochain/kangaroo-electron/blob/be753f055e4b35b9bd03a7092352a4775b0798bf/src/main/index.ts#L293-L300) upon quitting but should in principle also get killed automatically if the main nodejs process exists I think.

Reply:
Now I see that you're quitting the app only when that option was selected. All good